satpy/tests/compositor_tests/test_viirs.py

```diff
@@ -223,6 +242,254 @@ def test_hncc_dnb(self):
+                                            71.10179768327806, 71.33161009169649, 78.81291424983952])
+
+    def test_reflectance_corrector_viirs(self):
+        import xarray as xr
+        import dask.array as da
+        import numpy as np
+        import datetime
+        from satpy.composites.viirs import ReflectanceCorrector
+        from satpy import DatasetID
+        ref_cor = ReflectanceCorrector(dem_filename='_fake.hdf', optional_prerequisites=[
+         DatasetID(name='satellite_azimuth_angle', wavelength=None, resolution=371, polarization=None, calibration=None,
+                   level=None, modifiers=None),
+         DatasetID(name='satellite_zenith_angle', wavelength=None, resolution=371, polarization=None, calibration=None,
+                   level=None, modifiers=None),
+         DatasetID(name='solar_azimuth_angle', wavelength=None, resolution=371, polarization=None, calibration=None,
+                   level=None, modifiers=None),
+         DatasetID(name='solar_zenith_angle', wavelength=None, resolution=371, polarization=None, calibration=None,
+                   level=None, modifiers=None)],
+                                       name='I01', prerequisites=[], wavelength=(0.6, 0.64, 0.68), resolution=371,
+                                       calibration='reflectance', modifiers=('sunz_corrected_iband',
+                                                                             'rayleigh_corrected_crefl_iband'),
+                                       sensor='viirs')
+
+        self.assertEqual(ref_cor.attrs['modifiers'], ('sunz_corrected_iband', 'rayleigh_corrected_crefl_iband'))
+        self.assertEqual(ref_cor.attrs['calibration'], 'reflectance')
+        self.assertEqual(ref_cor.attrs['wavelength'], (0.6, 0.64, 0.68))
+        self.assertEqual(ref_cor.attrs['name'], 'I01')
+        self.assertEqual(ref_cor.attrs['resolution'], 371)
+        self.assertEqual(ref_cor.attrs['sensor'], 'viirs')
+        self.assertEqual(ref_cor.attrs['prerequisites'], [])
+        self.assertEqual(ref_cor.attrs['optional_prerequisites'], [
+            DatasetID(name='satellite_azimuth_angle', wavelength=None, resolution=371, polarization=None,
+                      calibration=None, level=None, modifiers=None),
+            DatasetID(name='satellite_zenith_angle', wavelength=None, resolution=371, polarization=None,
+                      calibration=None, level=None, modifiers=None),
+            DatasetID(name='solar_azimuth_angle', wavelength=None, resolution=371, polarization=None, calibration=None,
+                      level=None, modifiers=None),
+            DatasetID(name='solar_zenith_angle', wavelength=None, resolution=371, polarization=None, calibration=None,
+                      level=None, modifiers=None)
+            ])
+
+        area, dnb = self.data_area_ref_corrector()
+
+        def make_xarray(wavelength, modifiers, calibration, file_type, file_key, name, standard_name, units):
```

Comment:
Maybe it is github's weird formatting but this looks like it needs a `self` as the first argument.